### PR TITLE
Arachne 1995/multi atlas

### PR DESCRIPTION
**Pull request by pavgra**

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    @Column
+    @Enumerated(value = EnumType.STRING)
+    private AtlasAuthSchema authType;
+
+    @Column
+    private String username;
+
+    @Column
+    private String password;
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java]: password encryption?

File: src/main/java/com/odysseusinc/arachne/datanode/service/impl/AtlasServiceImpl.java [hunk 2]
```diff
@@ -32,76 +36,191 @@
...
 
+        if (atlas.getAuthType() != null) {
+            existing.setAuthType(atlas.getAuthType());
+        }
+
+        if (atlas.getUsername() != null) {
+            existing.setUsername(atlas.getUsername());
+        }
+
+        if (atlas.getPassword() != null) {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/datanode/service/impl/AtlasServiceImpl.java]: not sure. but what if password masquerade?

File: src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        Atlas atlas = (Atlas) o;
+
+        return getId() != null ? getId().equals(atlas.getId()) : atlas.getId() == null;
+    }
+
+    @Override
+    public int hashCode() {
+
+        return getId() != null ? getId().hashCode() : 0;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java]: 0

File: src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
+package com.odysseusinc.arachne.datanode.model.atlas;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/model/atlas/Atlas.java]: copyright

File: src/main/java/com/odysseusinc/arachne/datanode/dto/converters/BaseAtlasEntityToCommonEntityDTOConverter.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.odysseusinc.arachne.datanode.dto.converters;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/converters/BaseAtlasEntityToCommonEntityDTOConverter.java]: copyright

File: src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/AtlasDetailedDTO.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package com.odysseusinc.arachne.datanode.dto.atlas;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/AtlasDetailedDTO.java]: copyright

File: src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/AtlasDTO.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package com.odysseusinc.arachne.datanode.dto.atlas;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/AtlasDTO.java]: copyright

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AtlasController.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    @RequestMapping(value = "/api/v1/atlases", method = GET)
+    public Page<AtlasDTO> list(
+            @PageableDefault(page = 1)
+            @SortDefault.SortDefaults({
+                    @SortDefault(sort = "name", direction = Sort.Direction.ASC)
+            }) Pageable pageable
+    ) {
+
+        Pageable search = new PageRequest(pageable.getPageNumber() - 1, pageable.getPageSize(), pageable.getSort());
+        Page<Atlas> atlasList = atlasService.findAll(search);
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AtlasController.java]: it is not list

File: src/main/java/com/odysseusinc/arachne/datanode/controller/AtlasController.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
+package com.odysseusinc.arachne.datanode.controller;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/controller/AtlasController.java]: copyright

File: src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/BaseAtlasEntity.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
+package com.odysseusinc.arachne.datanode.dto.atlas;
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/datanode/dto/atlas/BaseAtlasEntity.java]: copyright

**Pull request merged**